src/Data/Aeson/Types/ToJSON.hs

```diff
@@ -1291,6 +1273,7 @@ instance ToJSON1 Maybe where
 
 instance (ToJSON a) => ToJSON (Maybe a) where
     toJSON = toJSON1
+    omitField = null
     toEncoding = toEncoding1
```

Comment:
A `Maybe a` will be omitted when (1) `omitNothingFields` is `True`, and (2) when the maybe is empty.

This preserves the current `ToJSON` behavior of `Maybe a` without needing overlapping instances.